src/test/java/com/microsoft/sqlserver/jdbc/resultset/ResultSetTest.java

```diff
@@ -73,4 +77,27 @@ public void testJdbc41ResultSetMethods() throws Exception {
             con.close();
         }
     }
+
+    /**
+     * Tests ResultSet#isWrapperFor and ResultSet#unwrap.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testTesultSetWrapper() throws SQLException {
+        try (Connection con = DriverManager.getConnection(connectionString);
+             Statement stmt = con.createStatement()) {
+            
+            stmt.executeUpdate("create table " + tableName + " (col1 int, col2 text, col3 int identity(1,1) primary key)");
+            
+            try (ResultSet rs = stmt.executeQuery("select * from " + tableName)) {
+                assertTrue(rs.isWrapperFor(ResultSet.class));
+                assertTrue(rs.isWrapperFor(ISQLServerResultSet.class));
+
+                assertSame(rs, rs.unwrap(ResultSet.class));
+                assertSame(rs, rs.unwrap(ISQLServerResultSet.class));
+            }
```

Comment:
Please drop the tables before the test finishes. It will throw exception that object already exists.